docroot/sites/classrooms.uiowa.edu/modules/classrooms_core/src/RoomItemProcessor.php

```diff
@@ -0,0 +1,218 @@
+          $entity_features = $entity->get('field_room_technology_features')->getString();
+          $entity_features = explode(', ', $entity_features);
+          // Sort lists before comparing.
+          sort($entity_features);
+          sort($tech_features);
+          if ($entity_features !== $tech_features) {
+            $updated = TRUE;
+            $entity->set('field_room_technology_features', $tech_features);
+          }
+        }
+      }
+    }
+    return $updated;
+  }
+
+  /**
+   * Returns a single record for the entity.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The entity to pull a record for.
+   *
+   * @return array|mixed
+   *   The record or an empty array.
+   */
+  public static function getRecord(ContentEntityInterface $entity): mixed {
+    // Grab MAUI room data.
+    $building_id = $entity
+      ->field_room_building_id
+      ?->target_id;
+    $room_id = $entity
+      ->field_room_room_id
+      ?->value;
+    /** @var \Drupal\uiowa_maui\MauiApi $maui_api */
+    $maui_api = \Drupal::service('uiowa_maui.api');
+    $results = $maui_api->getRoomData($building_id, $room_id);
+    // The record is returned inside the first entry of the $data array. Return
+    // this if it exists, or an empty array.
+    return $results[0] ?? [];
+  }
+
+  /**
+   * Process the record before comparison.
+   */
+  public function processRecord(&$record) {
+    // Set to null if we don't have a proper int.
+    if (isset($record->maxOccupancy) && filter_var($record->maxOccupancy, FILTER_VALIDATE_INT) === FALSE) {
+      $record->maxOccupancy = NULL;
+    }
+    // Account for the room name that is only a single space.
+    if (isset($record->roomName) && strlen($record->roomName) <= 1) {
+      $record->roomName = NULL;
+    }
+    if (isset($record->roomType)) {
+      $record->roomType = $this->processRecordTerm($record->roomType, 'room_types', TRUE);
+    }
+    if (isset($record->acadOrgUnitName)) {
+      $record->acadOrgUnitName = $this->processRecordTerm($record->acadOrgUnitName, 'units', TRUE);
+    }
+    if (isset($record->regionList)) {
+      $query = \Drupal::entityQuery('taxonomy_term')
+        ->condition('vid', 'scheduling_regions')
+        ->execute();
+
+      if ($query) {
+        $storage = \Drupal::entityTypeManager()->getStorage('taxonomy_term');
+        $terms = $storage->loadMultiple($query);
+        // If we weren't able to map it, we have scheduling regions
+        // that we don't want to display, so we'll want to set the
+        // regionList to null.
+        $region = NULL;
+        foreach ($terms as $term) {
+          if ($api_mapping = $term->get('field_api_mapping')?->value) {
+            if (in_array($api_mapping, $record->regionList)) {
+              // If we found a mappable region, set it.
+              $region = $term->id();
+              break;
+            }
+          }
+        }
+        $record->regionList = $region;
```

Comment:
```suggestion
        $record->regionList = NULL;
        foreach ($terms as $term) {
          if ($api_mapping = $term->get('field_api_mapping')?->value) {
            if (in_array($api_mapping, $record->regionList)) {
              // If we found a mappable region, set it.
              $record->regionList = $term->id();
              break;
            }
          }
        }
```
I think this should work without creating a variable.